src/dsql/parse.y

```diff
@@ -5095,11 +5095,19 @@ decimal_keyword
 
 %type <legacyField> float_type
 float_type
-	: FLOAT precision_opt
+	: FLOAT precision_opt_nz
 		{
+		    // Precision is binary digits of the significand: 1-24 for 32 bit single precision, 25-53 for 64 bit double precision
+			// Precision 0 is the 'no precision specified' case, which defaults to 32 bit single precision
+			SLONG precision = $2;
+
+			if (precision != 0 && (precision < 1 || precision > 53))
+				yyabandon(YYPOSNARG(2), -842, Arg::Gds(isc_precision_err2) << Arg::Num(1) << Arg::Num(53));
+																// Precision must be between 1 and 53
+
 			$$ = newNode<dsql_fld>();
```

Comment:
const SLONG, please. And maybe it's worth defining 53 and 24 as named constants in the beginning of this file.

Reply:
I followed what `decfloat_type` does. Should I change it there as well?

Reply:
Then leave your code as is. I will change both in a separate commit.

Reply:
Ok, I'll leave that as is.